src/NUnitEngine/nunit.engine/Runners/ProcessRunner.cs

```diff
@@ -150,21 +193,64 @@ protected override AsyncTestEngineResult RunTestsAsync(ITestEventListener listen
         /// <param name="force">If true, cancel any ongoing test threads, otherwise wait for them to complete.</param>
         public override void StopRun(bool force)
         {
-            _remoteRunner.StopRun(force);
+            try
+            {
+                _remoteRunner.StopRun(force);
+            }
+            catch (Exception e)
+            {
+                log.Error("Failed to stop the remote run. {0}", e.Message);
+            }
         }
 
         protected override void Dispose(bool disposing)
         {
             base.Dispose(disposing);
 
-            if (disposing && _agent != null)
+            try
+            {
+                if (disposing && _agent != null)
+                {
+                    log.Info("Stopping remote agent");
+                    _agent.Stop();
+                    _agent = null;
+                }
+            }
+            catch (Exception e)
             {
-                log.Info("Stopping remote agent");
-                _agent.Stop();
+                log.Error("Failed to stop the remote agent. {0}", e.Message);
                 _agent = null;
             }
         }
 
+        TestEngineResult CreateFailedResult(Exception e)
+        {
+            var suite = XmlHelper.CreateTopLevelElement("test-suite");
+            XmlHelper.AddAttribute(suite, "type", "Assembly");
+            XmlHelper.AddAttribute(suite, "id", TestPackage.ID);
+            XmlHelper.AddAttribute(suite, "name", TestPackage.Name);
+            XmlHelper.AddAttribute(suite, "fullname", TestPackage.FullName);
+            XmlHelper.AddAttribute(suite, "runstate", "NotRunnable");
+            XmlHelper.AddAttribute(suite, "testcasecount", "1");
+            XmlHelper.AddAttribute(suite, "result", "Failed");
+            XmlHelper.AddAttribute(suite, "label", "Error");
+            XmlHelper.AddAttribute(suite, "start-time", DateTime.UtcNow.ToString("u"));
+            XmlHelper.AddAttribute(suite, "end-time", DateTime.UtcNow.ToString("u"));
+            XmlHelper.AddAttribute(suite, "duration", "0.001");
+            XmlHelper.AddAttribute(suite, "total", "1");
+            XmlHelper.AddAttribute(suite, "passed", "0");
+            XmlHelper.AddAttribute(suite, "failed", "1");
+            XmlHelper.AddAttribute(suite, "inconclusive", "0");
+            XmlHelper.AddAttribute(suite, "skipped", "0");
+            XmlHelper.AddAttribute(suite, "asserts", "0");
+
+            var failure = suite.AddElement("failure");
+            var message = failure.AddElementWithCDataSection("message", e.Message);
+            var stack = failure.AddElementWithCDataSection("stack-trace", e.StackTrace);
+
+            return new TestEngineResult(suite);
+        }
+
```

Comment:
I feel like we may want to put this elsewhere at some point, or perhaps combine it with what we do in other cases. Good for the release though.